### DISCUSSION TITLE
Are there any weird hardware calls that are being made or is everything done through its respective API?

### DISCUSSION THREAD

**Discussion by ryan2458** [seerge/g-helper#312] [Q&A] [answered]
From what I've seen, an already written wrapper for the NVIDIA public API is used and then a wrapper was created in this project for the AMD public API for the gpu controls.

How are the fans being controlled?  Who is responsible for managing that API?

Are there any calls to control the hardware that's not done through a tried and tested API?  If so, how has that been tested?

**Comment by seerge** [comment, marked as answer]
@ryan2458 On laptops fans are controlled by asus bios, not via Nvidia / AMD API. 
There are known endpoints to apply an (optional) custom fan curve for cpu / gpu fans, that app uses (same as armoury or linux kernel for example)

I didn't get a your take in general here.

**Reply by ryan2458** [reply]
This answers my question.

I understood that the fans were not controlled by NVIDIA/AMD API.  I was wondering who controlled that, which you have clarified as being done through the bios.

I'm not a hardware programmer, so the main thing I was getting at with this question was to make sure that the known endpoints were actually being used and not some "hacky" way of controlling hardware, which you have also clarified.

Thank you.
